src/Microsoft.AspNet.OData.Shared/Query/Expressions/TransformationBinderBase.cs

```diff
@@ -0,0 +1,172 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics.Contracts;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using Microsoft.AspNet.OData.Common;
+using Microsoft.AspNet.OData.Interfaces;
+using Microsoft.OData.Edm;
+using Microsoft.OData.UriParser;
+
+namespace Microsoft.AspNet.OData.Query.Expressions
+{
+    internal class TransformationBinderBase : ExpressionBinderBase
+    {
+        internal TransformationBinderBase(ODataQuerySettings settings, IWebApiAssembliesResolver assembliesResolver, Type elementType,
+            IEdmModel model) : base(model, assembliesResolver, settings)
+        {
+            Contract.Assert(elementType != null);
+            _elementType = elementType;
+
+            this._lambdaParameter = Expression.Parameter(this._elementType, "$it");
+        }
+
+        protected Type _elementType;
+        protected ParameterExpression _lambdaParameter;
+        protected bool _classicEF = false;
```

Comment:
shouldn't really expose fields as protected/public but, if we do, should expose with standard (i.e., Pascal) naming.